File: csharp/bob/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        string trimmedStatement = statement.Trim();
        bool isEmpty = string.IsNullOrWhiteSpace(trimmedStatement);
        bool hasLetter = trimmedStatement.Any(char.IsLetter);
        bool isQuestion = trimmedStatement.EndsWith('?');
        bool isAllCaps = hasLetter && trimmedStatement.ToUpper() == trimmedStatement;

        return isEmpty
            ? "Fine. Be that way!"
            : isQuestion && isAllCaps
                ? "Calm down, I know what I'm doing!"
                : isAllCaps ? "Whoa, chill out!" : isQuestion ? "Sure." : "Whatever.";
    }

}

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  const speech = message.trim();
  if (speech === "") return `Fine. Be that way!`;

  const isQuestion = speech.endsWith(`?`);
  const isShoting = /[a-z]/i.test(speech) && speech === speech.toUpperCase();

  if (isShoting && isQuestion) return `Calm down, I know what I'm doing!`;
  if (isShoting) return `Whoa, chill out!`;
  if (isQuestion) return `Sure.`;

  return `Whatever.`;
};
